compiler/src/lntoamm/Microstatement.ts

```diff
@@ -1165,19 +1165,48 @@ ${letdeclarationAst.t} on line ${letdeclarationAst.line}:${
     if (val.statementType === StatementType.REREF) {
       val = Microstatement.fromVarName(val.alias, scope, microstatements);
     }
-    val.statementType = StatementType.LETDEC;
-    microstatements.push(
-      new Microstatement(
-        StatementType.REREF,
-        scope,
-        true,
-        val.outputName,
-        val.outputType,
-        [],
-        [],
-        letAlias,
-      ),
-    );
+    if (val.outputType === Type.builtinTypes.string) {
```

Comment:
Basically creates code that looks like:

```ln
const someUUID = 'my string';
let actualVarName = someUUID.clone();
```

Then the string constant is hoisted away and the let variable clones the global memory version to allow mutation.